fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricMaterialBuilder.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.block;
+
+import net.minecraft.block.Material;
+import net.minecraft.block.MaterialColor;
+import net.minecraft.block.piston.PistonBehavior;
+import net.minecraft.util.DyeColor;
+
+import net.fabricmc.fabric.mixin.object.builder.MaterialBuilderAccessor;
+
+public class FabricMaterialBuilder extends Material.Builder {
+	public FabricMaterialBuilder(MaterialColor color) {
```

Comment:
Something should probably be done about these constructors and factory methods, some use static of(), some static create(), some public constructors like this one. Full consistency is IMO not /that/ important, but it should at least be considered and discussed.

Reply:
Hmm I would be fine with using `of` or `create`.

Since this a builder we may want to consider that `of` or create doesn't make 100% sense

Reply:
how about... `builder`?

Reply:
`FabricMaterial.builder()` or something

Reply:
I'd personally lean towards `builder()`.

@sfPlayer1 what do you think of that?

Reply:
I don't like it, you use this in the non-builder class, which we don't have.

Reply:
well normally with a builder you either have a `builder` method in the built class or you use the builder's constructor. IMO the constructor is fine and should be the standard, at least for builders.

Reply:
Chose to stick with the ctors since this is relatively simple

Reply:
> 
> 
> Chose to stick with the ctors since this is relatively simple

No, there should be consistency for consturcting material builder and block settings. Either make material builder use utility methods or make block settings use constructors.

Reply:
Is it possible to have a simple util method to copy from an existing material

Reply:
Also really would like to have methods to unset settings, such as unset requires tools